src/query/Index.cpp

```diff
@@ -211,6 +211,25 @@ ProjectionMatch Index::findNearestProjection(const Eigen::VectorXd &location, in
   }
 }
 
+ProjectionMatch Index::findCellInterpolation(const Eigen::VectorXd &location, int n)
+{
+  if (_mesh->getDimensions() == 2) {
+    auto matchedTriangles = getClosestTriangles(location, n);
+    for (const auto &match : matchedTriangles) {
+      auto polation = mapping::Polation(location, _mesh->triangles()[match.index]);
+      if (polation.isInterpolation()) {
+        return {polation, 0.0};
+      }
+    }
+
+    // If no triangle is found, fall-back on NP
+    return findNearestProjection(location, n);
+  } else {
+    PRECICE_ASSERT(_mesh->getDimensions() == 3, "Volume coupling 3D not  implemented");
```

Comment:
This should really be `PRECICE_UNREACHABLE("Volume coupling 3D not implemented")`